### PR TITLE
Omit empty config fields and show regex upon re-marshalling to elide secrets

### LINKED ISSUE #862: Issues with marshaled config on status page
* Regular expressions are shown as `{}` instead of the actual regular expression.
* unset fields are shown as `null` or `""` or their default value

Raw config

```yaml
# ...
      routes:
      - match:
          alertname: MothershipHighErrorRate
        match_re: 
          controller: .*MediaController|.*StreamsController
        receiver: playback-crit
# ...
```

Printed config

```yaml
# ...
      routes:
      - receiver: playback-crit
        match:
          alertname: MothershipHighErrorRate
        match_re:
          controller:
            regexp: {}
# ...
```

### PR DESCRIPTION
**Pull request by Conorbro**
Fixes #862

### CODE REVIEW

File: config/notifiers.go [hunk 1]
```diff
@@ -235,17 +235,17 @@ func (c *SlackConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+	RoomID        string `yaml:"room_id,omitempty" json:"room_id,omitempty"`
+	From          string `yaml:"from,omitempty" json:"from,omitempty"`
+	Notify        bool   `yaml:"notify,omitempty" json:"notify,omitempty"`
+	Message       string `yaml:"message,omitempty" json:"message,omitempty"`
+	MessageFormat string `yaml:"message_format,omitempty" json:"message_format,omitempty"`
+	Color         string `yaml:"color,omitempty" json:"color,omitempty"`
 
 	// Catches all undefined fields and must be empty after parsing.
-	XXX map[string]interface{} `yaml:",inline" json:"-"`
+	XXX map[string]interface{} `yaml:",inline" ,omitemptyjson:"-"`
```
Comment (mxinden) [inline config/notifiers.go]: Is this a typo? I can't find `omitemptyjson` in the specs.

Comment (Conorbro) [inline config/notifiers.go]: Yep that's a typo!

**Comment by stuartnelson3** [comment]
@grobie can you give this a try?

**Comment by grobie** [comment]
Can we write a test instead?

**Comment by Conorbro** [comment]
Not sure about default values yet but this should fix and test for the issue regarding regexps printing as `{}` instead of the actual regular expression and unset fields shown as `""` or `null`.

**Comment by Conorbro** [comment]
@stuartnelson3 @grobie 

Any thoughts?

**Comment by stuartnelson3** [comment]
Disregard the comment about breakage, it was because your version of alertmanager was missing updates from mainline

**Pull request merged**